src/test/ui/consts/const-points-to-static.rs

```diff
@@ -0,0 +1,12 @@
+// compile-flags: -Zunleash-the-miri-inside-of-you
+
+#![allow(dead_code)]
+
+const TEST: &u8 = &MY_STATIC;
```

Comment:
This doesn't actually *access* the static, so it seems like `before_access_static` is misnamed...?

Reply:
Looks like `before_access_static` is called by `get_static_alloc`... but why does that get called here (and not just the function that fetches the *size* of the allocation)?

Reply:
This constant does not access the underlying static until validation. If you look at the error, you can see that it differs from `const-prop-read-static-in-const.rs`. Here it's a validation error, in the reading from a static test it's a const eval error.

I can make this a const eval error, but that would require to essentially run const validation on every `Operand::Constant` to detect lazy alloc ids (or at least a reduced version that just walks all the alloc ids and checks them for lazy ones).

Reply:
Ah, validation, that makes sense. I forgot we re-check other statics/consts that are referenced. (This can lead to quadratic complexity at least, so maybe we should reconsider that?)

Reply:
But also, this means we won't error when referencing statics in other crates, right? As validation stops at crate boundaries?

Reply:
Well.....

https://github.com/rust-lang/rust/blob/a916ac22b9f7f1f0f7aba0a41a789b3ecd765018/src/librustc_mir/interpret/validity.rs#L454-L456

As long as the reference to the other static is at the base with the correct type, we don't need to validate because the `const_eval` query will do this validation anyway. But everything else needs to get validated. The other validations can't get cached because that could cause cycles.

Reply:
But with this PR, this means we'll also "miss" references to statics in other crates, right? Like, `const-points-to-static.rs` but with `MY_STATIC` in a different crate?

Reply:
yea, that's true. As the FIXME says, this is not sound

Reply:
Well, actual reads will be checked, so it's not catastrophic either.